Assets/Mirror/Runtime/NetworkIdentity.cs

```diff
@@ -573,16 +573,11 @@ bool OnSerializeSafely(NetworkBehaviour comp, NetworkWriter writer, bool initial
             return result;
         }
 
-        // OnSerializeAllSafely is in hot path. caching the writer is really
-        // worth it to avoid large amounts of allocations.
-        static NetworkWriter onSerializeWriter = new NetworkWriter();
-
         // serialize all components (or only dirty ones if not initial state)
         // -> returns serialized data of everything dirty,  null if nothing was dirty
         internal byte[] OnSerializeAllSafely(bool initialState)
         {
-            // reset cached writer length and position
-            onSerializeWriter.SetLength(0);
+            NetworkWriter onSerializeWriter = NetworkWriter.GetPooledWriter();
```

Comment:
you should recycle it in line 590, which is a very common case, and not an error.  Otherwise you are leaving a lot of garbage behind

Reply:
Fixed